ui.apps/src/main/content/jcr_root/apps/asset-share-commons/clientlibs/clientlib-site/js/cart/event-handlers.js

```diff
@@ -0,0 +1,99 @@
+/*
+ * Asset Share Commons
+ *
+ * Copyright [2017]  Adobe
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*global jQuery: false, AssetShare: false, handleButtonsUpdateOnDetails, handleBulkCartButtonsUpdate, toggleCartButtons, flipAction  */
+jQuery((function($, ns, cart) {
+	"use strict";
+	var element,
+		ADD_TO_CART = "add-to-cart",
+		REMOVE_FROM_CART = "remove-from-cart",
+		EVENT_DETAILS_PAGE_LOAD = "asset-share-commons.details.load",
+		EVENT_SEARCH_END = "asset-share-commons.search.end",
+		EVENT_CART_CLEAR = "asset-share-commons.cart.clear",
+		EVENT_CART_ADD = "asset-share-commons.cart.add",
+		EVENT_CART_REMOVE = "asset-share-commons.cart.remove";
+	$("body").on(EVENT_DETAILS_PAGE_LOAD,function(event, actionButtons) {
+		element = actionButtons;
+		handleButtonsUpdateOnDetails();
```

Comment:
Pass in element

Reply:
Removed the method handleButtonsUpdateOnDetails() . Calling just one method handleBulkCartButtonsUpdate() based on add/remove assetshare ids